test-specifications/token-service/tokenDeleteTransaction.md

```diff
@@ -0,0 +1,87 @@
+
+- The ID of the token to delete.
+
+| Test no | Name                                                       | Input                                                                                               | Expected response                                                                   | Implemented (Y/N) |
+|---------|------------------------------------------------------------|-----------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------|-------------------|
+| 1       | Deletes an immutable token                                 | tokenId=<VALID_IMMUTABLE_TOKEN_ID>                                                                  | The token deletion fails with an TOKEN_IS_IMMUTABLE response code from the network. | N                 |
+| 2       | Deletes a mutable token                                    | tokenId=<VALID_MUTABLE_TOKEN_ID>, commonTransactionParams.signers=[<VALID_MUTABLE_TOKEN_ADMIN_KEY>] | The token deletion succeeds.                                                        | N                 |
+| 3       | Deletes a token that doesn't exist                         | tokenId="123.456.789"                                                                               | The token deletion fails with an INVALID_TOKEN_ID response code from the network.   | N                 |
+| 4       | Deletes a token with no token ID                           | tokenId=""                                                                                          | The token deletion fails with an SDK internal error.                                | N                 |
+| 5       | Deletes a token that was already deleted                   | tokenId=<DELETED_TOKEN_ID>, commonTransactionParams.signers=[<DELETED_TOKEN_ADMIN_KEY>]             | The token deletion fails with an TOKEN_WAS_DELETED response code from the network.  | N                 |
+| 6       | Deletes a token without signing with the token's admin key | tokenId=<VALID_MUTABLE_TOKEN_ID>                                                                    | The token deletion fails with an INVALID_SIGNATURE response code from the network.  | N                 |
+| 7       | Deletes a token but signs with an incorrect private key    | tokenId=<VALID_MUTABLE_TOKEN_ID>, commonTransactionParams.signers=[<INCORRECT_VALID_PRIVATE_KEY>]   | The token deletion fails with an INVALID_SIGNATURE response code from the network.  | N                 |
+
+#### JSON Request Example
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 64362,
+  "method": "deleteToken",
+  "params": {
+    "tokenId": "0.0.15432",
+    "commonTransactionParams": {
+      "signers": [
+        "302E020100300506032B657004220420DE6788D0A09F20DED806F446C02FB929D8CD8D17022374AFB3739A1D50BA72C8"
+      ]
+    }
+  }
+}
+```
+
+#### JSON Response Example
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 64362,
```

Comment:
I guess this is valid for other TCK specs as well.

Reply:
This is just the JSON-RPC request number https://www.jsonrpc.org/specification#request_object. It's a way for the client to associate the request it sends out with any response object it receives back from the server. The TCK increments this number with every request it sends out, it does not match up with the number of tests since a single test could send out multiple requests.